addons/web/controllers/main.py

```diff
@@ -1459,26 +1450,25 @@ def content_image(self, xmlid=None, model='ir.attachment', id=None, field='datas
     def _content_image(self, xmlid=None, model='ir.attachment', id=None, field='datas',
                        filename_field='name', unique=None, filename=None, mimetype=None,
                        download=None, width=0, height=0, crop=False, quality=0, access_token=None,
-                       placeholder=None, **kwargs):
+                       **kwargs):
         status, headers, image_base64 = request.env['ir.http'].binary_content(
             xmlid=xmlid, model=model, id=id, field=field, unique=unique, filename=filename,
             filename_field=filename_field, download=download, mimetype=mimetype,
             default_mimetype='image/png', access_token=access_token)
 
         return Binary._content_image_get_response(
             status, headers, image_base64, model=model, id=id, field=field, download=download,
-            width=width, height=height, crop=crop, quality=quality,
-            placeholder=placeholder)
+            width=width, height=height, crop=crop, quality=quality)
 
     @staticmethod
     def _content_image_get_response(
             status, headers, image_base64, model='ir.attachment', id=None,
             field='datas', download=None, width=0, height=0, crop=False,
-            quality=0, placeholder='placeholder.png'):
+            quality=0):
         if status in [301, 304] or (status != 200 and download):
             return request.env['ir.http']._response_by_status(status, headers, image_base64)
         if not image_base64:
-            if placeholder is None and model in request.env:
+            if model in request.env:
```

Comment:
You can completely delete this part of the code I think, `_get_placeholder_filename` should be completely removed now in favor of our new field when it is needed.